src/lib/sql/sql_translator.cpp

```diff
@@ -820,6 +819,19 @@ void SQLTranslator::_translate_select_list_groupby_having(const hsql::SelectStat
       }
     }
   }
+
+  // For SELECT DISTINCT, we add an aggregate node that groups by all output columns, but doesn't use any aggregate
+  // functions, e.g.: `SELECT DISTINCT a, b ...` becomes  `SELECT a, b ... GROUP BY a, b`.
+  //
+  // This might create unnecessary aggregate nodes when we already have an aggregation that creates unique results:
+  // `SELECT DISTINCT a, MIN(b) FROM t GROUP BY a` would have one aggregate that groups by a and calculates MIN(b), and
+  // one that groups by both a and MIN(b) without calculating anything. Fixing this would require us to check for all
+  // GROUP BY cases whether the results are guaranteed to be unique or not. Doable, but currently no priority.
```

Comment:
No wait, in the case of `SELECT DISTINCT a, MIN(b) FROM t GROUP BY a`, how would you get away with just one AggregateNode? Imo you need two.

Reply:
> imo, that would be optimizer concern.

The SQLTranslator should not add useless nodes and hope for the optimizer to catch them if it can help it.

> No wait, in the case of SELECT DISTINCT a, MIN(b) FROM t GROUP BY a, how would you get away with just one AggregateNode? Imo you need two.

No issue here - the `GROUP BY a` already ensures that `a` is DISTINCT, so we wouldn't need a second one.

Reply:
> The SQLTranslator should not add useless nodes and hope for the optimizer to catch them if it can help it.

It depends on how you define "help it". First and foremost I would aim for a clear separation of concerns, so the SQLTranslator should pretty stupidly perform AST->LQP translation. The optimizer should, well, optimize. 

Examples: We're not creating JOINs for join predicates in the where clause, we do not omit redundant Sorts.

Reply:
Ok, agreed. The translator should not do extra work just to avoid inefficient plans. Still, I think that a comment in places where it might do so helps. I'll change the comment so that it points to the optimizer.